### PR TITLE
feat: Update PAS Egress Vector Explainer for nullability

### PR DESCRIPTION
**Pull request by akundla-google**
Includes updates to language and examples

### CODE REVIEW

**Review by johncox-google — APPROVED** [review]
Change LGTM. Just some optional comments to consider before merge, no need for re-review.

Thanks for doing this!

File: bidding_auction_services_protected_app_signals_egress.md [hunk 1]
```diff
@@ -17,71 +17,102 @@ The wire representation of the `egressPayload` will be noised per feature type.
...
 Based on the information in this document, adtechs can write parsers to prepare the `egressPayload` and transform the values it contains into features for use in their model training systems. 
 
-
 # Feature types
 
 We support two kinds of feature types: primitives, which can contain a single feature value; and collections, which can contain multiple primitives.
 
+**Nullability**
+
+Every feature type can optionally be made `nullable`, in which case the value `null` can be represented. By default features are _not_ nullable. A null value for a type is indicated by adding a signaling bit, 0 (for null) or 1 (for non-null), to the least-significant bit of the serialized representation. Consequently, the wire representation for a feature with nullable = true is always 1 bit larger than with nullable = false.
```
Comment (salmanmlk) [inline bidding_auction_services_protected_app_signals_egress.md:26]: Consider quoting "nullable" and "null" with backticks consistently.

File: bidding_auction_services_protected_app_signals_egress.md [hunk 2]
```diff
@@ -17,71 +17,102 @@ The wire representation of the `egressPayload` will be noised per feature type.
...
+
+*   `nullable`: boolean indicating whether the `boolean-feature-type` can represent a null state or not. Defaults to `false`.
 
 **Wire format**
 
+Non-nullable:
+
 0 (`false`) or 1 (`true`)
 
+Null-able:
```
Comment (salmanmlk) [inline bidding_auction_services_protected_app_signals_egress.md:50]: s/Null-able/Nullable

Here and at other places

File: bidding_auction_services_protected_app_signals_egress.md [hunk 3]
```diff
@@ -17,71 +17,102 @@ The wire representation of the `egressPayload` will be noised per feature type.
...
 Parameters:
 
-
-
 *   `size`: unsigned integer indicating the number of bits based on the range of values.
 
-Expected value:  non-negative integer value in the range [0,2^<sup>size</sup>-1]
+      * Expected value:  non-negative integer value in the range [0,2^<sup>size</sup>-1]
 
-For example, if `size `= `3` `unsigned-integer-feature-type` can have a value : ` [0, 7] `and will occupy 3 bits on the wire.
```
Comment (salmanmlk) [inline bidding_auction_services_protected_app_signals_egress.md:56]: Here and at other places we talk about the size on the wire without taking nullability into account.
See if making this conditional on nullability would make it more clear.

**Review by salmanmlk — APPROVED** [review]
Thanks for the changes, Alek.

File: bidding_auction_services_protected_app_signals_egress.md [hunk 4]
```diff
@@ -90,54 +121,70 @@ These feature types represent a collection of homogeneous or heterogeneous value
...
-<strong>Wire format</strong>
+Examples:
 
-Sequential bit representation of `boolean-feature-type` values.
+*Non-nullable collection*:
 
-For example, if `size` = `4`, `bucket-feature-type` can have values which are of the type `boolean-feature-type`. If the `boolean-feature-type` values are [`true`, `false`, `true`, `false`] this will occupy `4` bits on the wire. Wire format will be `0101`
+If `size` = `4`, `bucket-feature-type` can have values which are of the type `boolean-feature-type`. If the `boolean-feature-type` values are [`true`, `false`, `true`, `false`] this will occupy `4` bits on the wire. Wire format will be `0101`
 
+Again for `size` = `4`, consider values which are of the type `boolean-feature-type`, this time with the values themselves all being `nullable = true`. If the `boolean-feature-type` values are [`true`, `false`, `true`, `null`] this will occupy `8` bits on the wire. Wire format will be `00 11 01 11` (spaces for readability).
```
[the remark this answers is not in the document]
Comment (akundla-google) [reply bidding_auction_services_protected_app_signals_egress.md:146]: Thanks John, I have an example like that for the histogram below

File: bidding_auction_services_protected_app_signals_egress.md (hunk 1 above)
Comment (akundla-google) [reply bidding_auction_services_protected_app_signals_egress.md:26]: I reviewed the doc and made one or two minor adjustments for consistency. When I use null and nullable as english words, they are left formatted like the rest of the prose. When used as an example of written code, or used to represent a value in null's case, they are backticked into appearing as code.

File: bidding_auction_services_protected_app_signals_egress.md (hunk 2 above)
Comment (akundla-google) [reply bidding_auction_services_protected_app_signals_egress.md:50]: good call. done.

File: bidding_auction_services_protected_app_signals_egress.md [hunk 5]
```diff
@@ -90,54 +121,70 @@ These feature types represent a collection of homogeneous or heterogeneous value
...
-*   <code>size<strong>: </strong></code>number of values in the bucket.
+Sequential bit representation of `boolean-feature-type` values.
 
-<strong>Wire format</strong>
+Examples:
 
-Sequential bit representation of `boolean-feature-type` values.
+*Non-nullable collection*:
 
-For example, if `size` = `4`, `bucket-feature-type` can have values which are of the type `boolean-feature-type`. If the `boolean-feature-type` values are [`true`, `false`, `true`, `false`] this will occupy `4` bits on the wire. Wire format will be `0101`
```
[the remark this answers is not in the document]
Comment (akundla-google) [reply bidding_auction_services_protected_app_signals_egress.md:112]: updated across the doc, thanks

File: bidding_auction_services_protected_app_signals_egress.md (hunk 3 above)
Comment (akundla-google) [reply bidding_auction_services_protected_app_signals_egress.md:56]: I may have misunderstood, but this set of updates is written as if nullability does not affect the value of the size param; that the size param represents the bits of data to be communicated independent of whether there is also a null bit or not. Is that wrong? I will have to make some minor edits if my understanding was incorrect.

**Review by akundla-google — COMMENTED** [review]
Thanks both, @salmanmlk PTAL as I think I may have misunderstood how nullability affects size.

File: bidding_auction_services_protected_app_signals_egress.md (hunk 3 above)
Comment (akundla-google) [reply bidding_auction_services_protected_app_signals_egress.md:56]: discussed offline, my original understanding was correct. Updated explainer language to increase clarity.

**Comment by akundla-google** [comment]
@salmanmlk ready for re-review

**Comment by salmanmlk** [comment]
> @salmanmlk ready for re-review

Thanks! LGTM

**Pull request merged**